pyqmc/reblock.py

```diff
@@ -0,0 +1,55 @@
+import pyblock
+import pandas as pd
+import numpy as np
+from scipy.stats import sem
+
+
+def reblock(data, reblocks, col):
+    '''
+        Reblocks data according to “Error estimates on averages of correlated data”,
+        H. Flyvbjerg, H.G. Petersen, J. Chem. Phys. 91, 461 (1989).
+    '''
+    edat = data[col].values
+    n = len(edat)
+    for i in range(reblocks):
+        edat_prime = []
+        for j in range(1, int(len(edat)/2+1)):
+            edat_prime.append((edat[2*j-2] + edat[2*j-1])/2)
+        edat = edat_prime
+    return np.array(edat)
+
+
+def optimally_reblocked(data, cols):
+    '''
+        Uses pyblock to find optimal reblocking of input data. Takes in pandas
+        DataFrame of raw data and selected columns to reblock, returns DataFrame
+        of reblocked data.
+    '''
+    stats = pyblock.pd_utils.reblock(data[cols])
+    reblocked_data = pd.DataFrame(columns=['mean', 'standard error', 'standard error error',
+                                           'reblocked_data'], index=cols)
+    for c in cols:
+        reblocked_data.loc[c] = pyblock.pd_utils.reblock_summary(stats[1][c]).squeeze()
+        reblocks = pyblock.pd_utils.optimal_block(stats[1][c])
+        reblocked_data.at[c, "reblocked_data"] = reblock(data, reblocks, c)
+    return reblocked_data
+
+
+def test_reblocking():
+    '''
+        Tests reblock function above against PyBlock values
+    '''
+
+    cols = ["energytotal", "energyee", "energyei", "energyke"]
+    #data = pd.read_csv("./data.csv")
+    data = pd.read_json("./dmcdata.json")
```

Comment:
Tests should not reference outside data. You should generate a distribution that you know the average and mean of, and then check that the error bars are correct.

Reply:
I created a data set based on the tutorial given by pyblock. I checked mean, standard error, and standard error error.

Reply:
The test won't run unless the data file is included..so how is someone else supposed to run it?